Merge upstream 5.7.17 into Percona Server

percona-ysorokin opened the pull request.

percona-ysorokin commented:
Jenkins build link (with '--big-test')
http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/592/

percona-ysorokin commented:
' File_io_test.SyncOnInvalidFileDescriptor' in 'merge_keyring_file_tests' unit test fails only when 'libeatmydata.so' is preloaded.

This became an issue after commit mysql/mysql-server@a3e501d
"Bug #23498254: KEYRING_FILE SHOULD NOT DUMP ITS CONTENT WHEN KEYRING FILE IS NOT AVAILABLE - fix for 5.7 based on file's stat"

laurynas-biveinis commented:
A test needing update for 32-bits:

main.bug80134                            w2 [ fail ]  Found warnings/errors in server log file!
        Test ended at 2017-01-23 14:01:55
line
2017-01-23T19:01:50.689505Z 0 [Warning] option 'sort_buffer_size': unsigned value 1125899906842624 adjusted to 4294967295
^ Found warnings in /mnt/workspace/mysql-5.7-param/BUILD_TYPE/debug/Host/centos6-32/build/mysql-test/var/2/log/mysqld.1.err
ok

laurynas-biveinis commented:
The unit test failure still has to handled, it's a regression. It is probably a bug in libeatmydata (it returns success for any file descriptor passed). So -
- A libeatmydata bug should be reported (https://github.com/stewartsmith/libeatmydata)
- Remove libeatmydata from unit tests until the bug is fixed and we deploy it (not sure how to do that - split Jenkins MTR test invocation so that actual MTR runs without unit tests with LD_PRELOAD, and then separately run unit tests without?) or disable the test

laurynas-biveinis commented:
OOM at http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/592/BUILD_TYPE=release,Host=debian-jessie-64bit/consoleText? Should be logged to JIRA

laurynas-biveinis commented:
Likewise for http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/592/BUILD_TYPE=release,Host=debian-wheezy-x64/consoleText

laurynas-biveinis commented:
Diff LGTM. Also consider rebasing on the current trunk for the merge conflicts.

percona-ysorokin commented:
'main.bug80134' MTR test case updated.

Also fixed problem with `DTRACE INSTRUMENTED` instead of `DTRACE_INSTRUMENTED` (no underscore) in 'storage/example/CMakeLists.txt ' discovered on CentOS 5.11 x86.

percona-ysorokin commented:
Verified that 'libeatmydata' issue exists in upstream even with the latest version of the library.

Submitted an issue to 'stewartsmith/libeatmydata'
https://github.com/stewartsmith/libeatmydata/issues/1

robgolebiowski commented on include/my_global.h:
```diff
@@ -281,13 +281,24 @@ typedef socket_len_t SOCKET_SIZE_TYPE; /* Used by NDB */
+static inline int is_directory_separator(char c)
+{
+#ifdef _WIN32
+  return c == FN_LIBCHAR || c == FN_LIBCHAR2;
```
I am just wondering if this was tested on windows. FN_LIBCHAR2 was removed from line 284 and I cannot find FN_LIBCHAR2 in msdn. Thus I assume this will not compile on windows.

percona-ysorokin replied:
it was removed from the Linux part of the `#ifdef`
`_WIN32` section still has it (line 275)

robgolebiowski replied:
Ack :)

percona-ysorokin commented:
Added a workaround for 'libeatmydata' issue to 'mysql-5.7-template' Jenkins build definition - unit tests are now run in a separate "./mtr ..." command without 'libeatmydata'.

Updated Jenkins build link
http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/601
